homeassistant/components/hue/sensor_base.py

```diff
@@ -194,13 +196,23 @@ def register_component(self, binary, async_add_entities):
             else:
                 new_sensors.append(current[api[item_id].uniqueid])
 
+        removed_items = await get_removed_devices(
+            self.hass,
+            self.config_entry,
+            [value.uniqueid for value in api.values()],
+            current,
+        )
+
         async_add_sensor_entities = self._component_add_entities.get(False)
         async_add_binary_entities = self._component_add_entities.get(True)
         if new_sensors and async_add_sensor_entities:
             async_add_sensor_entities(new_sensors)
         if new_binary_sensors and async_add_binary_entities:
             async_add_binary_entities(new_binary_sensors)
 
+        for item_id in removed_items:
+            del current[item_id]
```

Comment:
You could even go as far as removing these inside the helper method . Makes me even wonder how much of our update mechanism can be generalized.